pages/studentPortal/labs/multiplication-connect/Index.tsx

```diff
@@ -16,106 +25,124 @@ export const calculateWinner = (grid: GameBoardBlock[]) => {
-        rows[i + 2][index].isSelected &&
-        rows[i + 3][index].isSelected
-          ? console.log("(vertical) Four in a row!")
+        rows[i][index].selectedBy == player &&
+        rows[i + 1][index].selectedBy == player &&
+        rows[i + 2][index].selectedBy == player &&
+        rows[i + 3][index].selectedBy == player
+          ? console.log(player, "(vertical) Four in a row!")
           : "";
       }
     }
     if (i >= 3) {
       // Ascending diagonal check
       for (let index = 0; index < rows[i].length - 3; index++) {
-        rows[i][index].isSelected &&
-        rows[i - 1][index + 1].isSelected &&
-        rows[i - 2][index + 2].isSelected &&
-        rows[i - 3][index + 3].isSelected
-          ? console.log("(ascending diagonal) Four in a row!")
+        rows[i][index].selectedBy == player &&
+        rows[i - 1][index + 1].selectedBy == player &&
+        rows[i - 2][index + 2].selectedBy == player &&
+        rows[i - 3][index + 3].selectedBy == player
+          ? console.log(player, "(ascending diagonal) Four in a row!")
           : "";
       }
       // Descending diagonal check
       for (let index = 3; index < rows[i].length; index++) {
-        rows[i][index].isSelected &&
-        rows[i - 1][index - 1].isSelected &&
-        rows[i - 2][index - 2].isSelected &&
-        rows[i - 3][index - 3].isSelected
-          ? console.log("(descending diagonal) Four in a row!")
+        rows[i][index].selectedBy == player &&
+        rows[i - 1][index - 1].selectedBy == player &&
+        rows[i - 2][index - 2].selectedBy == player &&
+        rows[i - 3][index - 3].selectedBy == player
+          ? console.log(player, "(descending diagonal) Four in a row!")
           : "";
       }
     }
   }
 };
 
+const createGrid = () => {
+  let arr = [];
+  let newGrid = [];
+  for (let i = 4; i < 25; i++) i % 2 === 0 ? arr.push(i) : "";
+  for (let i = 0; i < 35; i++) {
+    let gridNumber = getRandomItemFromArray(arr);
+    newGrid.push({
+      id: i,
+      gridNumber: gridNumber,
+      selectedBy: selectedBy.Unselected,
+    });
+  }
+  return newGrid;
+};
+
 const Index: FC = () => {
   const [grid, setGrid] = useState([]);
   const [newGame, setNewGame] = useState(0);
+  const [isPlayerOne, setIsPlayerOne] = useState(true);
 
-  // initialize grid on page load
-  const createGrid = () => {
-    let arr = [];
-    let newGrid = [];
-    for (let i = 4; i < 25; i++) i % 2 === 0 ? arr.push(i) : "";
-    for (let i = 0; i < 35; i++) {
-      let gridNumber = getRandomItemFromArray(arr);
-      newGrid.push({
-        id: i,
-        gridNumber: gridNumber,
-        isSelected: false,
-      });
-    }
-
-    return newGrid;
-  };
-
-  // useEffect not triggering re-render of GameBoard even though grid State is being updated
-  // grid state may need to included in dependency array but this results in infinite loop of renders
   useEffect(() => {
     console.log(`newGame: ${newGame}`);
     setGrid(createGrid);
```

Comment:
Did you figure out why this effect wasn't working?

Reply:
Yeah I worked thorough it with Abhi and Jason and managed to figure it out. Jason got it working by adding another useEffect in GameBoardBlock to update the blocks when the data passed to them as a prop changes (on game reset). Bc the grid was updating but the individual block states in the child component wasn't.

But then he mentioned the implementation was kind of messy and he would skip the `block` state altogether and just manage the `grid` state instead. So I just reworked the code bc that made more sense.